ti: Add more firmware, GPU etc packages

jsuhaas22 opened the pull request:
# Description

_Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change._

TI Debian images support different features, such as IPC, Weston (as OOB screen), GPU acceleration, camera etc. So to have these working by-default in our images, this PR adds certain packages such as:
* gstreamer (required for camera)
* chromium
* GPU and mesa packages
* k3conf (a diagnostic tool for TI devices)
* weston ... etc

Additionally, there are 2 more things done here:
1. Weston service file is added, and placed in `/lib/systemd/system/`. This is to start weston on boot by-default. Note that weston doesn't launch on first boot when the armbian-firstlogin script runs, but starts launching from subsequent boots.
2. `dkms autoinstall` builds and installs the ti-img-rogue-<plat>-dkms package for the current kernel. If this isn't done here, it will done during first boot, which would make the first boot very slow.

# How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration._

- boot tested AM62x to see weston come up
- also performed other tests such as wayland-simple-egl and chromium to ensure GPU works

# Checklist:

_Please delete options that are not relevant._

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

jsuhaas22 committed:
ti: Add more packages

TI maintains a bunch of Debian packages for TI devices, such as a
diagnostic tool (k3conf), firmware, GPU out-of-tree modules and mesa.
Add these packages to the Debian image by-default.

The GPU package is built using dkms. Add an explicit step to bind dkms
packages to kernel. Currently the dkms package gets installed before the
kernel, therefore if this step is excluded, the binding would occur on
first boot, which would cause the first boot time to get longer.
Therefore add the "dkms autoinstall" step.

[0] https://www.debian.org/doc/manuals/maint-guide/dother.en.html#conffiles

Signed-off-by: Suhaas Joshi <s-joshi@ti.com>

glneo commented:
Starting Weston as a root systemd service was a hack we did for our Yocto SDK distro, it was not correct and we are switching to using a proper weston user and launching with a login-manager (Emptty). For Armbian you should do the same or re-use GDM or similar default manager to launch a weston session.

jsuhaas22 commented on config/sources/families/k3.conf:
```diff
@@ -20,13 +20,33 @@ declare -g SPD_OPTEED="SPD=opteed"
+	TI_PACKAGES+=("weston" "emptty")
+fi
+
+if [[ "${CHROMIUM_SUPPORT}" == "yes" ]] ; then
```
Currently, this is the case with AM62SIP and AM62LP. This is due to memory constraints (at least in case of SIP).

glneo commented on config/sources/families/k3.conf:
```diff
@@ -20,13 +20,33 @@ declare -g SPD_OPTEED="SPD=opteed"
 declare -g INSTALL_HEADERS="yes"
 
-declare -g TI_PACKAGES=()
+declare -g TI_PACKAGES=("k3conf" "vim")
```
`vim` is nice to have, but not really TI specific, if we want `vim` installed by default you should bubble this up to one of the top level package groups.

igorpecovnik replied:
yes, vim can be added to the default packages group.

jsuhaas22 replied:
Done

glneo commented on config/boards/sk-am62b.conf:
```diff
@@ -16,3 +16,7 @@ ATF_BOARD="lite"
 CC33XX_SUPPORT="yes"
+GPU_SUPPORT="yes"
+SOC_ID="am62"
+DISPLAY_SUPPORT="yes"
```
There is already `HAS_VIDEO_OUTPUT` which if not set to `no` is assumed to be true, you should be able to make use of that instead.

jsuhaas22 replied:
Done.

> assumed to be true

The files in lib/ just check if its set to "no", but don't check if its "yes" anywhere (i.e. they don't really initialize it anywhere to "yes"). So I have set it to "yes" explicitly in sk-am62b.conf.

jsuhaas22 replied:
Done

jsuhaas22 committed:
config: cli: Add `vim` to default list of packages

`vim` is pretty widely used, so include it in the list of default
packages to install in a CLI image.

Signed-off-by: Suhaas Joshi <s-joshi@ti.com>

glneo commented on config/sources/families/k3.conf:
```diff
@@ -20,13 +20,33 @@ declare -g SPD_OPTEED="SPD=opteed"
+if [[ "${GPU_SUPPORT}" == "yes" ]] ; then
+	TI_PACKAGES+=("ti-img-rogue-driver-${SOC_ID}-dkms" "ti-img-rogue-umlibs-${SOC_ID}" "ti-img-rogue-tools-${SOC_ID}" "ti-img-rogue-firmware-${SOC_ID}")
+	TI_PACKAGES+=("mesa-vulkan-drivers" "libgl1-mesa-dri")
+	TI_PACKAGES+=("gstreamer1.0-tools" "gstreamer1.0-plugins-good" "gstreamer1.0-plugins-bad" "gstreamer1.0-plugins-base")
```
What uses these? Shouldn't the user pull these in if they need them like other packages?

jsuhaas22 replied:
My rationale is: our boards are for evaluating and prototyping with our SoCs. If our SoC supports camera, then our image should ideally have by-default whatever it needs to run the camera. Our docs run the camera with GStreamer, which is why I added these. We had these in our previous flavour of Debian as well.

jsuhaas22 commented on config/sources/families/k3.conf:
```diff
@@ -20,13 +20,33 @@ declare -g SPD_OPTEED="SPD=opteed"
+fi
+
+if [[ "${CHROMIUM_SUPPORT}" == "yes" ]] ; then
+	TI_PACKAGES+=("chromium")
```
> My rationale is: our boards are for evaluating and prototyping with our SoCs. If our SoC supports camera, then our image should ideally have by-default whatever it needs to run the camera. Our docs run the camera with GStreamer, which is why I added these. We had these in our previous flavour of Debian as well.

Same answer, but adapted to showing off GPU acceleration with browser. We had this in previous Debian, and also have it as a demo in our docs.

But I can remove both GStreamer and Chromium, if you disagree with this

glneo commented on config/sources/families/k3.conf:
```diff
@@ -20,13 +20,28 @@ declare -g SPD_OPTEED="SPD=opteed"
+	TI_PACKAGES+=("weston" "emptty")
+fi
+
+TI_PACKAGES+=("firmware-cnm-wave" "firmware-ti-ipc-${SOC_ID}" "firmware-ti-connectivity")
```
We should fix up the packaging to that `firmware-ti-ipc` is not SoC specific. The files are small and do not conflict with each other, better to just have a common package, that is more inline with the "Debian" way of packaging.

leggewie replied:
good point!

if you want to push some files to a package armbian-ti-common in APA, just let me know.

glneo commented on config/boards/sk-am64b.conf:
```diff
@@ -16,3 +16,6 @@ SERIALCON="ttyS2"
 OPTEE_PLATFORM="k3-am64x"
+SOC_ID="am64"
+EXTRA_BOARD_PACKAGES=("firmware-ti-prueth-am64" "firmware-ti-pruhsr-am64" "firmware-ti-prusw-am64")
+HAS_VIDEO_OUTPUT="no"
```
Need newline after.

glneo commented:
> For now, based on internal discussions, I have removed the weston and emptty. It was decided that we should explore ways to introduce these as a separate desktop environment or something similar.
> 
> So from now, this PR is for adding extra packages related to vim, firmware, kernel modules etc.
> 
> Therefore I have also edited the PR heading.

The first patch still include `make weston the OOB` in the commit message, remove that.
